oauth_dropins/reddit.py

```diff
@@ -0,0 +1,111 @@
+
+  Provides methods that return information about this user and make OAuth-signed
+  requests to the Tumblr API. Stores OAuth credentials in the datastore. See
+  models.BaseAuth for usage details.
+
+  reddit-specific details: implements "access_token," which is really a refresh_token
+  see: https://stackoverflow.com/questions/28955541/how-to-get-access-token-reddit-api
+  The datastore entity key name is the reddit username.
+  """
+  # access token
+  refresh_token = ndb.StringProperty(required=True)
+  
+  def site_name(self):
+    return 'reddit'
+
+  def user_display_name(self):
+    """Returns the username.
+    """
+    return self.key_id()
+
+  def access_token(self):
+    """Returns the OAuth refresh token.
+    """
+    return self.refresh_token
+
+
+class StartHandler(handlers.StartHandler):
+  """Starts reddit auth. goes directly to redirect. passes to_path in "state"
+  """
+  NAME = 'reddit'
+  LABEL = 'Reddit'
+
+  def redirect_url(self, state=None):
+    if not state:
+      state = str(randint(100000,999999))
+    assert REDDIT_APP_KEY and REDDIT_APP_SECRET, \
+      "Please fill in the reddit_app_key and reddit_app_secret files in your app's root directory."
+    reddit = praw.Reddit(client_id=REDDIT_APP_KEY,
+                         client_secret=REDDIT_APP_SECRET,
+                         redirect_uri=self.request.host_url + self.to_path,
+                         user_agent='oauth-dropin reddit identity checker')
+    
+    # store the state for later use in the callback handler
+    models.OAuthRequestToken(id=state,
+                             token_secret=state,
+                             state=state).put()
+    return reddit.auth.url(['identity'], state+':'+self.to_path, 'permanent')
+
+  @classmethod
+  def button_html(cls, *args, **kwargs):
+    return super(cls, cls).button_html(
+      *args,
+      input_style='background-color: #CEE3F8; padding: 10px',
+      **kwargs)
+
+
+class CallbackHandler(handlers.CallbackHandler):
+  """OAuth callback. Only ensures that identity access was granted.
+  """
+
+  def get(self):
+    error = self.request.get('error')
+    st = self.request.get('state')
+    state = st.split(':')[0]
+    to_path = st.split(':')[1]
+    code = self.request.get('code')
+    if error or not state or not code:
+      # user declined
+      self.finish(None)
+      return
+    
+    # look up the stored state to check authenticity
+    request_token = models.OAuthRequestToken.get_by_id(state)
+    if request_token is None:
+      raise exc.HTTPBadRequest('Invalid oauth_token: %s' % request_token_key)
+
+    reddit = praw.Reddit(client_id=REDDIT_APP_KEY,
+                         client_secret=REDDIT_APP_SECRET,
+                         redirect_uri=self.request.host_url + to_path,
+                         user_agent='oauth-dropin reddit identity checker')
+
+    refresh_token = reddit.auth.authorize(code)
+    user_id = str(reddit.user.me())
+    
+    auth = RedditAuth(id=user_id,
```

Comment:
we generally also fetch the user's profile or other basic info, if possible, and store that in the auth object too, usually in a `user_json` property. mind doing that? can be in another PR. example:

https://github.com/snarfed/oauth-dropins/blob/afe5cc946e2d969a37d83e2a093195c1d5cf27db/oauth_dropins/mastodon.py#L357-L362

(the first `MastodonAuth` instantiation there is just a temporary convenience to reuse its `get()` method that includes the access token. feel free to structure yours however you want.)

Reply:
i cherry picked a few attributes, when i try to take them all it starts making requests to get all comments and posts.